src/LibVLCSharp/Helpers/MarshalExtensions.cs

```diff
@@ -166,5 +166,29 @@ internal static IntPtr ToUtf8(this string? str)
                 MarshalUtils.LibVLCFree(ref nativeString);
             return Encoding.UTF8.GetString(buffer, 0, buffer.Length);
         }
+
+#if !APPLE
+        /// <summary>
+        /// https://github.com/dotnet/runtime/blob/c4b9dabec8186a0d61f0cc3ea0b7efea579bf24e/src/libraries/System.Private.CoreLib/src/System/IO/Stream.cs#L720-L734
+        /// </summary>
+        /// <param name="stream"></param>
+        /// <param name="buffer"></param>
+        /// <returns></returns>
+        internal static int Read(this System.IO.Stream stream, Span<byte> buffer)
```

Comment:
How does it relates to the `stream.Read(Span<byte>)` method, and how come does it not cause any conflict for platforms where it's available ? We should really use the instance method where available, because it's optimized for some special cases (doesn't need the ArrayPool allocation) : See [here](https://github.com/dotnet/runtime/blob/c4b9dabec8186a0d61f0cc3ea0b7efea579bf24e/src/libraries/System.Private.CoreLib/src/System/IO/MemoryStream.cs#L367) and [there](https://github.com/dotnet/runtime/blob/c4b9dabec8186a0d61f0cc3ea0b7efea579bf24e/src/libraries/System.Private.CoreLib/src/System/IO/FileStream.Windows.cs#L412)

Reply:
Maybe the instance method is not yet available in the TFM we target? Would you be okay to add higher TFMs ? (like netstandard2.1 ? : https://docs.microsoft.com/en-us/dotnet/api/system.io.stream.read?view=netstandard-2.1 )

Reply:
https://github.com/videolan/libvlcsharp/blob/25dce8e43f0cb6568ca6b363fb4279e9ff0c62ed/src/LibVLCSharp/LibVLCSharp.csproj#L31 we already target 2.1

Reply:
This extension method was added because that signature was not available in some cases (on the Stream type, Span is available everywhere thanks to the nuget package). I'll modify it to have it call the regular implementation when possible.